include/lldb/Target/SwiftLanguageRuntime.h

```diff
@@ -326,9 +326,9 @@ class SwiftLanguageRuntime : public LanguageRuntime {
   /// Determines wether \c variable is the "self" object.
   static bool IsSelf(Variable &variable);
 
-  void AddToLibraryNegativeCache(const char *library_name);
+  void AddToLibraryNegativeCache(std::string library_name);
```

Comment:
Why not a StringRef?

Reply:
Because the negative cache is a std::unordered_set<std::string>. I suppose an llvm::StringMap would be better but I have to limit the scope of this patch somewhere.

Reply:
I think we should just replace the negative cache and the positive cache with a single one taking an `Optional<>` [also for MangledName -> Type]

Reply:
Fair enough. In that case it could still take a const ref, right?